include/swift/Reflection/TypeRefBuilder.h

```diff
@@ -919,13 +922,16 @@ class TypeRefBuilder {
 private:
   std::vector<ReflectionInfo> ReflectionInfos;
 
-  /// Index of the next Reflection Info that should be processed.
-  /// This assumes that Reflection Infos are never removed from the vector.
-  size_t FirstUnprocessedReflectionInfoIndex = 0;
-    
+  /// Reflection Infos we've already processed.
+  llvm::DenseSet<const ReflectionInfo *> ProcessedReflectionInfos;
```

Comment:
This will fail if `ReflectionInfos` ever reallocates, since it stores the structs by value and they'll move. I imagine that could happen if you're debugging a process that `dlopens` something containing Swift. We never _remove_ entries from `ReflectionInfos`, so it should be safe to make this a set of indexes instead.

Reply:
Good point!